dask/dataframe/partitionquantiles.py

```diff
@@ -264,7 +264,13 @@ def percentiles_to_weights(qs, vals, length):
         return ()
     diff = np.ediff1d(qs, 0.0, 0.0)
     weights = 0.5 * length * (diff[1:] + diff[:-1])
-    return vals.tolist(), weights.tolist()
+    return list(vals), weights.tolist()
+
+
+def replace_na(arr):
+    """Replace NA in array with np.nan"""
+    original_type = type(arr)
+    return original_type([np.nan if pd.isna(x) else x for x in arr])
```

Comment:
I don't have a great work-around in mind just yet. However, the GPU-CI failures seems to be the case that `x` is a single-element `cupy` array (like `array(2.)`), and calling `pd.isna` on a cupy array is technically not allowed.

Reply:
Oooh, thanks @rjzamora. Is there any way I could test things out without a GPU?